app/controllers/api_controller/providers.rb

```diff
@@ -79,9 +85,12 @@ def create_provider(data)
     end
 
     def edit_provider(provider, data)
+      check_connection_configurations(data)
+      merge_connection_configurations(provider, data) if data.key?("connection_configurations_array")
```

Comment:
Consider moving this condition into `merge_connection_configurations()` so it does nothing otherwise.